controllers/hosting/hostingdeployment_controller_test.go

```diff
@@ -1058,15 +1149,28 @@ func ExpectRequestToDeleteHostingDeployment(req interface{}, deployment *hosting
 	Expect(req).To(BeAssignableToTypeOf((*sagemaker.DeleteEndpointInput)(nil)))
 
 	deleteRequest := req.(*sagemaker.DeleteEndpointInput)
-	Expect(*deleteRequest.EndpointName).To(Equal(controllercommon.GetGeneratedJobName(deployment.ObjectMeta.GetUID(), deployment.ObjectMeta.GetName(), 63)))
+
+	var endpointName string
+	if deployment.Spec.EndpointName != nil {
+		endpointName = *deployment.Spec.EndpointName
+	} else {
+		endpointName = controllercommon.GetGeneratedJobName(deployment.ObjectMeta.GetUID(), deployment.ObjectMeta.GetName(), 63)
+	}
+	Expect(*deleteRequest.EndpointName).To(Equal(endpointName))
 }
 
 // Helper function to verify that the specified object is n UpdateEndpointInput and that it requests to update the HostingDeployment correctly.
 func ExpectRequestToUpdateHostingDeployment(req interface{}, deployment *hostingv1.HostingDeployment, expectedEndpointConfigName string) {
 	Expect(req).To(BeAssignableToTypeOf((*sagemaker.UpdateEndpointInput)(nil)))
 
 	updateRequest := req.(*sagemaker.UpdateEndpointInput)
-	Expect(*updateRequest.EndpointName).To(Equal(controllercommon.GetGeneratedJobName(deployment.ObjectMeta.GetUID(), deployment.ObjectMeta.GetName(), 63)))
+	var endpointName string
```

Comment:
Does this validate the case if the spec had no endpointname originally but is updated to have one and re-applied ?